src/ui.rs

```diff
@@ -88,13 +112,21 @@ impl Ui {
     pub fn stdout_formatter<'a>(&'a self) -> Box<dyn Formatter + 'a> {
         match &self.output {
             UiOutput::Terminal { stdout, .. } => self.new_formatter(stdout.lock()),
+            UiOutput::Paged { .. } => {
```

Comment:
We already have `self.output.child` available here (just replace `..` by `child`), and we can get the compiler to check that the panic in `pager_stdin()` is impossible if we instead get the stdin from that.

Actually, maybe we should even store the `ChildStdout` and `ChildStderr` in separate fields of `UiOutput::Paged`? Creating it something like this:

```
...
let child_stdout = child.stdout.take().unwrap();
let child_stderr = child.stderr.take().unwrap();
UiOutput::Paged {
  child,
  child_stdout,
  child_stderr,
}
```

Would that work? Then we also avoid unwrapping them on every call.

Reply:
Eh, I mean only `child_stdin`, but you probably understood that :)

Reply:
I'm not sure I understand the point of storing the child's `stdout` and `stderr`, since we don't interact with them.
Storing the child's `stdin` makes a lot of sense though. Let me see if the borrow checker will let me, I vaguely remember that being an obstacle before.

Reply:
Ah, we've discussed this before - the problem is that the `UiOutput::Paged` would hold both the `child` and `child.stdin`, which isn't allowed by ownership rules (I think?). The solution we came up with was to dynamically get a reference to the stdin (aka what we do here).

Reply:
`child.stdin.take().unwrap()` takes over the ownership from `child`, so I think that should work.

Reply:
Oh neat, https://doc.rust-lang.org/std/process/struct.Child.html#structfield.stdin uses the same trick.
This doesn't work for the formatter though, which stores a reference to the stdin, but I can inline a shortened version of this function there.